source/extensions/filters/network/thrift_proxy/config.cc

```diff
@@ -96,12 +97,22 @@ ProtocolType ProtocolOptionsConfigImpl::protocol(ProtocolType downstream_protoco
   return (protocol_ == ProtocolType::Auto) ? downstream_protocol : protocol_;
 }
 
+SINGLETON_MANAGER_REGISTRATION(thrift_route_config_provider_manager);
+
 Network::FilterFactoryCb ThriftProxyFilterConfigFactory::createFilterFactoryFromProtoTyped(
     const envoy::extensions::filters::network::thrift_proxy::v3::ThriftProxy& proto_config,
     Server::Configuration::FactoryContext& context) {
-  std::shared_ptr<Config> filter_config(new ConfigImpl(proto_config, context));
+  std::shared_ptr<Router::RouteConfigProviderManager> route_config_provider_manager =
+      context.singletonManager().getTyped<Router::RouteConfigProviderManager>(
+          SINGLETON_MANAGER_REGISTERED_NAME(thrift_route_config_provider_manager), [&context] {
+            return std::make_shared<Router::RouteConfigProviderManagerImpl>(context.admin());
+          });
+
+  std::shared_ptr<Config> filter_config(
+      new ConfigImpl(proto_config, context, *route_config_provider_manager));
 
-  return [filter_config, &context](Network::FilterManager& filter_manager) -> void {
+  return [route_config_provider_manager, filter_config,
```

Comment:
nit: comment here noting that we capture `route_config_provider_manager` here to prevent the reference to it passed into filter_config from being destroyed prematurely.